src/cascadia/TerminalApp/AppLogic.idl

```diff
@@ -62,5 +62,6 @@ namespace TerminalApp
         event Windows.Foundation.TypedEventHandler<Object, Object> FocusModeChanged;
         event Windows.Foundation.TypedEventHandler<Object, Object> FullscreenChanged;
         event Windows.Foundation.TypedEventHandler<Object, Object> AlwaysOnTopChanged;
+        event Windows.Foundation.TypedEventHandler<Object, Microsoft.Terminal.TerminalControl.SetTaskbarProgressEventArgs> SetTaskbarProgress;

```

Comment:
So here's a weird thing! (and it's going to seem like an unnecessary nit so I'm sorry about this)

Stylistically, we've been trying to adopt an event model of "Some thing Foo knows that a property Bar changed, so it raises a useless event". Then, when Baz, who's listening for that event on Foo gets the BarChanged event, it asks Foo for the current value of Bar, rather than using the value in the payload of the BarChanged event. 

You can see similar logic with the `FocusModeChanged`, `FullscreenChanged`, etc events above - the second object, the event, is usually nullptr, or some other useless object. When the handler gets the `FullscreenChanged` callback, actually just asks "what's your current Fullscreen state?"

We do this because it sometimes helps debounce events - You could imagine the progress changing from 0 -> 10, then 10 -> 20, then the window might get the first ProgressChanged event. In that case, it's usually best to just skip straight to the 20 state. 

If that makes sense 😬